### PR TITLE
Function/filter slides

### PR DESCRIPTION
**Pull request by pieplu**
Vanillate filterSlides
- Add utils, `filterNodeUtil`, than reproduce `jQuery.filter()`
- Add test for filterSlide (and unFilterSlide to detect filterSlide side-effect)

### CODE REVIEW

File: slick/slick.js [hunk 1]
```diff
@@ -2933,6 +2942,41 @@ Issues: http://github.com/kenwheeler/slick/issues
...
+			};
+		} else if (typeof filter === "function") {
+			filterFunction = filter;
+		} else {
+			filterFunction = function() {
+				return true;
+			};
+		}
+
+		return Array.prototype.slice.call(collectionOfNode).filter(filterFunction);
```
Comment (hs0ucy) [inline slick/slick.js]: What about using `return Array.prototype.filter.call(collectionOfNode, filterFunction);`? It will be more concise.

File: slick/slick.js [hunk 2]
```diff
@@ -2933,6 +2942,41 @@ Issues: http://github.com/kenwheeler/slick/issues
...
 
+	// @param  {Array}|{NodeList} `collectionOfNode` A collection of node to filter
+	// @param  {function}|{cssSelector}|{Node} `filter` The filter param
+	// @return {Array} of HTMLElement filtred by `filter`
+	// @usage Slick.filterNodeUtil(collectionOfNode, (funtion(elem,index,array){...} | '.my-selector' | node) );
+	// Equivalent to jQuery.filter() method, BUT do not support extended jQuery selector https://api.jquery.com/category/selectors/jquery-selector-extensions/
+	Slick.prototype.filterNodeUtil = function(collectionOfNode, filter) {
+		var _ = this,
+			domParent = document.createElement("div"),
+			selectedNodes,
```
Comment (hs0ucy) [inline slick/slick.js]: selectNodes = [],

File: slick/slick.js [hunk 3]
```diff
@@ -2933,6 +2942,41 @@ Issues: http://github.com/kenwheeler/slick/issues
...
+			filterFunction;
+
+		if (typeof filter === "string" && filter) {
+			Array.prototype.forEach.call(collectionOfNode, function(elem) {
+				domParent.appendChild(elem);
+			});
+			selectedNodes = _.queryAll(filter, domParent);
+			filterFunction = function(elem) {
+				return selectedNodes.lastIndexOf(elem) >= 0;
+			};
```
Comment (hs0ucy) [inline slick/slick.js:2959]: There's a less verbose version (of above):

```
var _ = this,           
            selectedNodes = [],
            filterFunction;

        if (typeof filter === "string" && filter) {
            Array.prototype.forEach.call(collectionOfNode, function(elem) {             
                if(_.matches(elem, filter)) {                  
                    selectedNodes.push(elem);
                }               
            });

            filterFunction = function(elem) {
                return selectedNodes.lastIndexOf(elem) >= 0;
            };
```

**Comment by pieplu** [comment]
... Strange for the test, they are the same, but only one fail for a timeout. I hope it's a "material" maters...

File: slick/slick.js [hunk 4]
```diff
@@ -2929,6 +2938,39 @@ Issues: http://github.com/kenwheeler/slick/issues
...
+	// @usage Slick.filterNodeUtil(collectionOfNode, (funtion(elem,index,array){...} | '.my-selector' | node) );
+	// Equivalent to jQuery.filter() method, BUT do not support extended jQuery selector https://api.jquery.com/category/selectors/jquery-selector-extensions/
+	Slick.prototype.filterNodeUtil = function(collectionOfNode, filter) {
+		var _ = this,
+			domParent = document.createElement("div"),
+			filterFunction;
+
+		if (typeof filter === "string" && filter) {
+			Array.prototype.forEach.call(collectionOfNode, function(elem) {
+				domParent.appendChild(elem); // Needed for matches, if collectionOfNode is out the DOM
```
Comment (pieplu) [inline slick/slick.js]: TODO, add on domParent, only if out the dom ...

**Comment by pieplu** [comment]
@hs0ucy correct done :)

File: slick/slick.js [hunk 5]
```diff
@@ -1005,22 +1005,28 @@ Issues: http://github.com/kenwheeler/slick/issues
...
+
 		if (filter !== null) {
 
 			_.$slidesCache = _.$slides;
 
 			_.unload();
 
-			_.$slideTrack.children(this.options.slide).detach();
+			_.filterNodeUtil(_.$slideTrack.get(0).children, _.options.slide).forEach(function(elem){
+				_.$slideTrack.get(0).removeChild(elem);
```
Comment (pieplu) [inline slick/slick.js:1016]: #36 @hs0ucy  maybe you detach method will be better
